### PR TITLE
Require static imports for Preconditions and Verify

### PR DESCRIPTION
**Pull request by kokosing**
Require static imports for Preconditions and Verify

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+Airbase 103
+
+* Checkstyle update:
+  - Require static imports for methods from com.google.common.base.Preconditions 
+  and com.google.common.base.Verify
```
Comment (findepi) [inline CHANGES.md:5]: indent

File: airbase-policy/src/main/resources/checkstyle/airbase-checks.xml [hunk 2]
```diff
@@ -78,6 +78,14 @@
         <property name="format" value="^[ \t]*import com\.google\.common\.base\.MoreObjects;$" />
         <property name="message" value="com.google.common.base.MoreObjects should only be used with static imports" />
     </module>
+    <module name="RegexpMultiline">
+        <property name="format" value="^[ \t]*import com\.google\.common\.base\.Preconditions;$" />
+        <property name="message" value="com.google.common.base.Preconditions should only be used with static imports" />
```
Comment (findepi) [inline airbase-policy/src/main/resources/checkstyle/airbase-checks.xml:83]: The caveat is that a caller class cannot define its own `checkArgument` method and then use Preconditions.
I think I am OK with this.

File: CHANGES.md [hunk 3]
```diff
@@ -1,3 +1,9 @@
+Airbase 103
+
+* Checkstyle update:
```
Comment (findepi) [inline CHANGES.md:3]:
```suggestion
* Checkstyle updates:
```

File: airbase-policy/src/main/resources/checkstyle/airbase-checks.xml (hunk 2 above)
Comment (kokosing) [reply airbase-policy/src/main/resources/checkstyle/airbase-checks.xml:83]: Yes. I would find such custom method very misleading.

**Comment by kokosing** [comment]
Comments addressed. Thanks!

File: airbase-policy/src/main/resources/checkstyle/airbase-checks.xml (hunk 2 above)
Comment (martint) [reply airbase-policy/src/main/resources/checkstyle/airbase-checks.xml:83]: This is actually a problem I've run into with String.format. I wanted to have a custom format() method that delegated to String.format, but that kind of usage is blocked. Very annoying.

Comment (kokosing) [reply airbase-policy/src/main/resources/checkstyle/airbase-checks.xml:83]: > I wanted to have a custom format() method that delegated to String.format, but that kind of usage is blocked. Very annoying.

I think one might say something opposite when noticed that the `format` got overriden. This surely depends on the case and person.

As with everything this is a tradeoff. https://github.com/prestosql/presto/pull/4733 showed that it was not a big deal in the code base. However, having this checkstyle rule makes a code review process a bit smoother which I find nice. Some people might find nit-picking comments a bit offensive. Thanks to this change we could avoiding discussions on obvious things.

File: CHANGES.md [hunk 4]
```diff
@@ -1,3 +1,9 @@
+Airbase 103
+
+* Checkstyle updates:
+  - Require static imports for methods from com.google.common.base.Preconditions
```
Comment (electrum) [inline CHANGES.md:4]: Let's shorten this to 

> Require static imports for methods from Preconditions and Verify

**Comment by kokosing** [comment]
Updated.
